Entities/Characters/Builder/BuilderLogic.as

```diff
@@ -142,26 +142,44 @@ bool RecdHitCommand(CBlob@ this, CBitStream@ params)
 
 	if (blobID == 0)
 	{
-		// block
 		CMap@ map = getMap();
 		if (map !is null)
 		{
-			uint16 type = map.getTile(tilepos).type;
-			if (!inNoBuildZone(map, tilepos, type))
+			if (map.getSectorAtPosition(tilepos, "no build") is null)
```

Comment:
What's the difference between using `inNoBuildZone` and checking for the no build map sectors here?  
And assuming this is the fix for backwall destruction part: why exactly does this bug happen and how does this fix it? Is there no better way? Might be worth documenting as comments in the code what the intents and possible sequence of events are.

Reply:
no, this is legacy code that i forgot to change -- sections of builderlogic that used getsectoratposition were replaced in last patch and i forgot to do it here  
no idea why exactly it happens, it's just that with fast building if you hold LMB+RMB at once while placing wooden doors you will be able to damage the backwall behind them pretty consistently, and this prevents that